pkg/identity/cache/cache.go

```diff
@@ -163,8 +159,6 @@ func (w *identityWatcher) watch(events allocator.AllocatorEventChan) {
 					default:
 						// Ignore modify events
 					}
-				case <-w.stopChan:
-					return
```

Comment:
Why should the identityWatcher continue to emit events into the policy subsystem if the agent is shutting down and the CachingIdentityAllocator is `Close()`d?

Reply:
Hmm, it doesn't look like that `Close()` function was called from the daemon yet anyway :thinking: This may have more to do with the tests than the runtime operation of the agent.

Reply:
Still, it would be nice to include the rationale for this aspect, since it seems like this moves to a pattern of:

```
foo := NewCachingIdentityAllocator(...) // set up the local identity allocator
foo.InitIdentityAllocator(...) // set up the global identity allocator, start handling those events
foo.Close() // Close the identity allocator but keep handling events from the global identity source
```

Reply:
ehh hmm, not sure I've quite got the logic right here. Core question I'm thinking about is how we ensure that the identityWatcher gets properly stopped. I'm going to put this down for now. Not sure the right answer here, maybe there's some answer around stopping the identityWatcher by closing its events channel from the sender through some other codepath.

Reply:
Yes, Close() is currently called only from daemon test code, but nonetheless it still deletes the global identity allocator, so events from there should stop coming. So this change essentially makes the local identity allocator never go away, which is in line with making locally allocated identities more persistent.

We could remove the Close() altogether, i suppose, to make this less confusing. The way I see it that when local identity allocator was added the code was patterned like for the global one. For local identity restoration we need the local allocator initialized earlier than the global one can be initialized (it needs node IP info for kvstore prefix). I figured making the initialization of the local allocator more "static" is the simplest approach, as it does not have the same constraints as the global allocator.